test/helpers/expect_successful_process_result.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:io';
+
+import 'package:mason/mason.dart';
+import 'package:test/test.dart';
+
+Future<ProcessResult> expectSuccessfulProcessResult(
```

Comment:
I think a matcher suits better, see https://github.com/VeryGoodOpenSource/r13n/blob/4c4a61180c77f4419758e5791afe36828e367fc5/e2e/test/r13n_brick/r13n_brick_test.dart#LL108.

Maybe we can have Very Good Matchers.

Reply:
Nice idea; we just have to add a flag to ignore std err from the validation and add stdout to the exception message. 
Ex1: npm install (😒) has the non-standard behavior of adding things to stderr and finishing the process with zero.
Ex2: `dart test` can fail but relevant information is present on stdout

Reply:
That sounds good to me! Shall we refactor all these `expectSuccessfulProcessResult` for matcher instead? The matcher can be a helper for the time being.

Reply:
I would leave that feature to its own pr (since it wont be minimal). Lets focus on unblocking main.

Reply:
Shall we open an issue with describing this refactor?